tests/lib/rules/no-side-effects-wait-for.test.ts

```diff
@@ -0,0 +1,210 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-side-effects-wait-for';
+
+const ruleTester = createRuleTester({
+  ecmaFeatures: {
+    jsx: true,
+  },
+});
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
+      code: `
+        import { waitFor } from '@testing-library/react';  
```

Comment:
Could you add some extra test to check that a `waitFor` imported from a module not related to testing-library is always valid please? Something like:

```javascript
import { waitFor } from '@somewhere/else';

// valid as this waitFor is not related to Testing Library
await waitFor(() => {
  expect(b).toEqual('b')
  fireEvent.keyDown(input, {key: 'ArrowDown'})
});
```